src/app/calendar-app/runbox-calendar-event.spec.ts

```diff
@@ -40,8 +51,244 @@ describe('RunboxCalendarEvent', () => {
+        expect(sut.toIcal()).not.toContain('INTERVAL', 'ical interval not set if default value (1)');
+        expect(sut.toIcal()).toContain('BYMONTHDAY=2', 'ical has BYMONTHDAY params');
+        expect(sut.toIcal()).toContain('BYDAY=MO', 'ical has BYDAY params');
+    });
+
+    it('should be possible to recur yearly on Xth day of the month', () => {
+        const sut = new RunboxCalendarEvent(
+          'testcal/testev', new ICAL.Event(new ICAL.Component(['vcalendar', [], [
+                [ 'vevent', [
+                    [ 'dtstart', {}, 'date',  moment().toISOString().split('T')[0] ],
+                    [ 'dtend',   {}, 'date',  moment().toISOString().split('T')[0] ],
+                    [ 'summary', {}, 'text',  'One-time event' ],
+                ] ]
+          ]])), ICAL.Time.fromJSDate(new Date()), ICAL.Time.fromJSDate(new Date()),
+        );
+        // Every 2nd of February of every year
+        sut.recurringFrequency = 'YEARLY';
+        sut.recurInterval = 1; // the default
+        sut.recursByMonth = ['2'];
+        sut.recursByMonthDay = ['2'];
+        expect(sut.recurringFrequency).toBe('YEARLY', 'recurrence can be retrieved');
+        expect(sut.toIcal()).toContain('RRULE:FREQ=YEARLY', 'recurrence seems to be stringified properly');
+        expect(sut.toIcal()).not.toContain('INTERVAL', 'ical interval not set if default value (1)');
+        expect(sut.toIcal()).toContain('BYMONTHDAY=2', 'ical has BYMONTHDAY params');
+        expect(sut.toIcal()).toContain('BYMONTH=2', 'ical has BYMONTH params');
+    });
+
+    it('should be possible to add a special case to a recurring event', () => {
+        const sut = new RunboxCalendarEvent(
+            'testcal/testev', new ICAL.Event(new ICAL.Component([
+                'vevent', [
+                  [ 'dtstart', {}, 'date',  '2021-01-25' ],
+                    [ 'dtend',   {}, 'date',  '2021-01-25' ],
+                    [ 'summary', {}, 'text',  'Weekly event' ],
+                    [ 'uid',     {}, 'text',  'unittestcase1' ],
+                    [ 'rrule',   {}, 'recur', { 'freq': 'WEEKLY' } ],
+                ]
+            ])), ICAL.Time.fromDateString('2021-02-01'), ICAL.Time.fromDateString('2021-02-01'),
+        );
+
+        const future = moment('2021-01-25').add(1, 'week').add(3, 'day');
+        // Feb 4th
+        sut.updateEvent(
+            future,
+            future,
+            sut.calendar,
+            RecurSaveType.THIS_ONLY,
+            'Moved weekly event', undefined, undefined,
+            true,
+            sut.recurringFrequency,
+            sut.recurInterval,
+            undefined, undefined, undefined, // and optional params..
+        );
+
+        expect(sut.toIcal()).toContain('SUMMARY:Moved weekly event');
+        expect(sut.toIcal()).toContain('RECURRENCE-ID;VALUE=DATE:20210201');
+        expect(sut.toIcal()).toContain('DTSTART;VALUE=DATE:20210204');
+        // expect(events.length).toBe(2);
```

Comment:
Are those a TODO to be fixed, or a leftover?

Reply:
Leftover *removes*